gp2-stress-experiment.ipynb

```diff
@@ -0,0 +1,933 @@
+    "        filtered_var[\"spectrum\", spec_num].values = y\n",
+    "\n",
+    "    return filtered_var\n",
+    "\n",
+    "if apply_median_filter:\n",
+    "    for k in wavelength.keys():\n",
+    "        if k == \"reference\":\n",
+    "            continue\n",
+    "\n",
+    "        wavelength[k] = median_filter(dataset=wavelength, variable_key=k)\n",
+    "\n",
+    "    # TODO export_to_tiff_stack for each"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "wav_x_vals = wavelength[\"sample\"].coords[\"wavelength\"].values\n",
+    "start_wave = np.max(wav_x_vals[:, 0])\n",
+    "stop_wave = np.min(wav_x_vals[:, -1])\n",
+    "bin_width_wave = np.mean(wav_x_vals[:, 1:]-wav_x_vals[:, :-1])\n",
+    "wavelength_rebin_params = sc.Variable([\"wavelength\"], values=np.arange(start_wave, stop_wave, bin_width_wave,\n",
+    "                                                           dtype=np.float64))\n",
+    "\n",
+    "rebinned_wavelength = sc.rebin(wavelength, \"wavelength\", wavelength_rebin_params)\n",
+    "\n",
+    "# if True, the whole initial state is integrated, if False initial state is also grouped\n",
+    "integrating_of_sample_in_inital_state = True\n",
+    "if integrated_neighbour_spectra:\n",
+    "    group_size = (324.0  / nx_target) * (324.0 / ny_target)\n",
+    "    factor = group_size\n",
+    "    rebinned_wavelength[\"sample\"] *= factor\n",
+    "\n",
+    "# Data matches here"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def weighted_sum(variable : sc.Variable, dim_to_sum):\n",
+    "    variable = variable.copy()\n",
+    "\n",
+    "    variances = sc.Variable(dims=variable.dims, values=variable.variances)\n",
+    "    d = sc.Dataset()\n",
+    "    d['data'] = variable / variances\n",
+    "    d['norm'] = sc.reciprocal(variances)\n",
+    "\n",
+    "    d.masks['zero_error'] = sc.equal(variances, 0.0 * variances.unit)\n",
+    "    \n",
+    "    d = sc.sum(d, dim_to_sum)\n",
+    "    sum = d['data']/d['norm']\n",
+    "    return sum\n",
+    "\n",
+    "# If we are not interested in strain regions for the unloaded sample, we can just combine all spectra to improve statistics.\n",
+    "# maybe another masking before summation, with tougher threshold to exclude boderline pixels.\n",
+    "rebinned_wavelength[\"sample\"] = weighted_sum(rebinned_wavelength[\"sample\"], \"spectrum\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "grouped = sc.Dataset()\n",
+    "grouped[\"sample_elastic\"] = sc.groupby(rebinned_wavelength[\"sample_elastic\"], \"detector_mapping\").sum(\"spectrum\")"
```

Comment:
Without having looked further down, why use `Dataset` if there is only one entry? Use `DataArray` in that case.